Platform/AlderlakeBoardPkg/Script/GpioDataConfig.py

```diff
@@ -12,46 +12,68 @@
 # in a platform specific Gpio library.
 #
 
+# Silicon/AlderlakePkg/Include/Register/PchRegsGpioVer2.h
 grp_info_lp = {
     # Grp     Index
     'GPP_B' : [ 0x0],
     'GPP_T' : [ 0x1],
     'GPP_A' : [ 0x2],
     'GPP_R' : [ 0x3],
-    'GPP_S' : [ 0x6],
+    'GPD'   : [ 0x4],
+    'GPP_S' : [ 0x5],
```

Comment:
Hi, the original commit had this table updated based on Ver4.h, or GpioSiLib.c tables 'mPchLpGpioGroupInfo' and 'mAdlPchSGpioGroupInfo'. Could you please double check whether we have to use Ver4 or Ver2.

If we have to use Ver2, then this patch set is correct, but GpioSiLib.c has to change.
If not, GpioSiLib.c is correct, and  this file should not be touched except for adding a dictionary for ADL-N.

Reply:
Code uses Ver2 mapping and 'mPchLpGpioGroupInfo' also uses Ver2 mapping. In this file original patch, 'grp_info_lp' dictionary didnt consider SPI as No register (0xFF), instead it counted as 0x4 and hence GPP_S was counted up as 0x6 which is incorrect. It should actually be 0x5 for GPP_S.

But one thing that doesnt match between mPchLpGpioGroupInfo and PchRegsGpioVer2.h is with **GPP_C** and **GPP_F**. According to mPchLpGpioGroupInfo, if we count, GPP_C should be 0xA and GPP_F should be 0xB. But if we follow PchRegsGpioVer2.h, GPP_C is defined as 0xB while GPP_F is defined as 0xA.

For this tool to work correctly, should we match with mPchLpGpioGroupInfo or the definitions for these groups from the header file? Clearly, they are not completely matching up.

Reply:
I syned with Sai again on this topic. It seems like for this tool to work correctly, we need to follow the indexing as per the mPchLpGpioGroupInfo[] from GpioSiLib.c, and not what is defined in PchRegsGpioVer2.h for Lp series. Similarly for S and N series. And SPI should still be considered as one of the indexes to be counted in. Please change these dictionaries according to the GpioGroupInfo[]. If SPI is not needed, skip adding it to the dictionary here but count it as +1.